hepcrawl/spiders/infn_spider.py

```diff
@@ -0,0 +1,229 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of hepcrawl.
+# Copyright (C) 2015, 2016 CERN.
+#
+# hepcrawl is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+"""Spider for INFN."""
+
+from __future__ import absolute_import, print_function
+
+from urlparse import urljoin
+
+import datetime
+import requests
+
+from scrapy.http import Request
+from scrapy.spiders import XMLFeedSpider
+
+from ..items import HEPRecord
+from ..loaders import HEPLoader
+from ..utils import get_temporary_file
+
+from ..dateutils import format_date
+
+
+class InfnSpider(XMLFeedSpider):
+
+    """INFN crawler
+    Scrapes theses metadata from INFN web page.
+    http://www.infn.it/thesis/index.php
+
+    1. If not local html file given, `get_list_file` gets one using POST
+       requests. Year is given as a argument, default is current year.
+
+    2. parse_node() iterates through every record on the html page.
+
+    3. If no pdf links are found, request to scrape the splash page is returned.
+
+    4. In the end, a HEPRecord is built.
+
+
+    Example usage:
```

Comment:
We are now using Sphinx documentation language in docs. So for the examples like this, can you:

1. Wrap in examples in `    .. code-block:: console`:

```python
.. code-block:: console

    scrapy crawl infn
    etc..

```

2. Then add the spider to the spider documentation in docs/api.rst:

```
.. automodule:: hepcrawl.spiders.infn_spider
   :members:
```